src/plugin/message/Subscriber/MessageSubscriber.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * This file is part of the Claroline Connect package.
+ *
+ * (c) Claroline Consortium <consortium@claroline.net>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Claroline\MessageBundle\Subscriber;
+
+use Claroline\CoreBundle\Entity\Log\MessageLog;
+use Claroline\CoreBundle\Event\CatalogEvents\MessageEvents;
+use Claroline\MessageBundle\Manager\MessageManager;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Contracts\EventDispatcher\Event;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class MessageSubscriber implements EventSubscriberInterface
+{
+    private $messageManager;
+    private $em;
+    private $security;
+    private $translator;
+
+    public function __construct(
+        MessageManager $messageManager,
+        EntityManagerInterface $em,
+        Security $security,
+        TranslatorInterface $translator
+    ) {
+        $this->messageManager = $messageManager;
+        $this->em = $em;
+        $this->security = $security;
+        $this->translator = $translator;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            MessageEvents::MESSAGE_SENDING => 'onMessageSending',
+        ];
+    }
+
+    public function onMessageSending(Event $event, string $eventName)
+    {
+        $this->messageManager->sendMessage(
+            $event->getContent(),
+            $event->getObject(),
+            $event->getReceivers(),
+            $event->getSender(),
+            $event->getWithMail()
+        );
+
+        $sender = $event->getSender() ?? $this->security->getUser();
+
+        foreach ($event->getReceivers() as $user) {
```

Comment:
I think you may have `Group` in this array. so `$logEntry->setReceiver($user);` will not work.

Reply:
Yes thx, I fixed It